### PR TITLE
 templates: add priority class `system-node-critical` to etcd pod

### PR DESCRIPTION
**Pull request by abhinavdahiya**
using `system-node-critical` from https://docs.okd.io/latest/admin_guide/scheduling/priority_preemption.html#admin-guide-priority-preemption-priority-class

### CODE REVIEW

File: templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml [hunk 1]
```diff
@@ -120,6 +120,9 @@ contents:
           containerPort: 2379
           protocol: TCP
       hostNetwork: true
+      priorityClassName: system-node-critical
```
Comment (smarterclayton) [inline templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml]: @sjenning are there any other magic incantations missing from etcd here?

**Comment by cgwalters** [comment]
Looks like this needs a rebase.  (I am still tempted to change the template unit tests to only sanity check one or two generated files, not all of them)

**Comment by crawford** [comment]
I tried testing this locally and I'm still seeing kubelet preempt etcd-member. Either my test procedure is faulty or this isn't sufficient.

**Comment by abhinavdahiya** [comment]
> I tried testing this locally and I'm still seeing kubelet preempt etcd-member. Either my test procedure is faulty or this isn't sufficient.

@sjenning on a local cluster that doesn't have this change, the etcd pod was evicted.

We edited the etcd static pod to includes these changes on the node and restarted the kubelet. But the kubelet still evicted the etcd pod...

File: templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml (hunk 1 above)
Comment (abhinavdahiya) [reply templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml]: from https://docs.okd.io/latest/admin_guide/scheduling/priority_preemption.html#admin-guide-priority-preemption-priority-class
```
System-cluster-critical - This priority class has a value of 2000000000 (two billion) and is used with pods that are important for the cluster. Pods with this priority class can be evicted from a node in certain circumstances. For example, pods configured with the system-node-critical priority class can take priority. However, this priority class does ensure guaranteed scheduling. Examples of pods that can have this priority class are fluentd, add-on components like descheduler, and so forth.
```

seems like cluster-critical can be evicted...

**Comment by abhinavdahiya** [comment]
rate limiting errors [e2e-aws](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/353/pull-ci-openshift-machine-config-operator-master-e2e-aws/1549)
```console
level=warning msg="Found override for ReleaseImage. Please be warned, this is not advised"
level=info msg="Consuming \"Install Config\" from target directory"
level=info msg="Creating cluster..."
level=error
level=error msg="Error: Error applying plan:"
level=error
level=error msg="2 errors occurred:"
level=error msg="\t* module.vpc.aws_route_table_association.worker_routing[0]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.worker_routing.0: timeout while waiting for state to become 'success' (timeout: 5m0s)"
level=error
level=error
level=error msg="\t* module.vpc.aws_route_table_association.route_net[5]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.route_net.5: timeout while waiting for state to become 'success' (timeout: 5m0s)"
level=error
level=error
level=error
level=error
level=error
level=error msg="Terraform does not automatically rollback in the face of errors."
level=error msg="Instead, your Terraform state file has been partially updated with"
level=error msg="any resources that successfully completed. Please address the error"
level=error msg="above and apply again to incrementally change your infrastructure."
level=error
level=error
level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply using Terraform"
```

will retest in a bit.

**Comment by ashcrow** [comment]
```
clusterversion.config.openshift.io/version condition met
/bin/bash: line 52: /etc/passwd: Permission denied
```

**Pull request merged**